src/core/session.ts

```diff
@@ -154,8 +154,8 @@ export class Session implements FormSubmitObserverDelegate, HistoryDelegate, Lin
 
   // Navigator delegate
 
-  allowsVisitingLocation(location: URL) {
-    return this.applicationAllowsVisitingLocation(location)
+  allowsVisitingLocationWithAction(location: URL, action?: Action) {
+    return this.locationWithActionIsSamePage(location, action) || this.applicationAllowsVisitingLocation(location)
```

Comment:
We allow the visit, it's just a silent one. Running the `locationWithActionIsSamePage` check first, prevents the `turbo:before-visit` event from being triggered in `applicationAllowsVisitingLocation`